spotifylib/spotifylib.py

```diff
@@ -313,19 +314,15 @@ def _retrieve_token(session, user, payload):
             LOGGER.exception(response.content)
             raise SpotifyError("Couldn't get new token from refresh token. "
                                "Got: {}".format(response.content))
-
-        values = [response.json().get(key) for key in Token._fields]
-        if not values[3]:
-            # in case of refresh, we don't get the refresh token back so we will
-            # just inject it. Of course we should already have it since this is
-            # a refresh request so session should already be set up maybe this
-            # can be a little simpler, hm....
-            values[3] = session.token.refresh_token
-        if not all(values):
+        tokens = response.json()
+        if not tokens.get('refresh_token', {}):
```

Comment:
you don't need the empty dictionary here as default argument. This will return None without it which is just fine. Also I think that it would be good to have an inline comment on why we do this since after 6 months nobody will remember that the refresh does not return the refresh token. This is slightly subtle so it should be documented.